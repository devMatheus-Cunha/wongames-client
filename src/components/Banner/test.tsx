import { screen } from '@testing-library/react'

import Banner from '.'
import { renderWithTheme } from '../../utils/tests/helpers'

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(
      <Banner
        title="Defy death"
        subtitle="Play the new CrashLands season"
        img="https://source.unsplash.com/user/willianjusten/1042x580"
        buttonLabel="Buy now"
        buttonLink="/games/defy-death"
      />
    )

    expect(
      screen.getByRole('heading', { name: /Defy death/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Play the new CrashLands season/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('img', { name: /Defy death/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
