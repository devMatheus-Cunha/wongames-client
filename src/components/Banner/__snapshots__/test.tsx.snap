// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c0 {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(180deg, #ff5f5f 0%, #f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  text-decoration: none;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c0:hover {
  background: linear-gradient(180deg, #e35565 0%, #d958a6 50%);
}

<main
  class=""
>
  <div
    aria-label="Defy death"
    class=""
    role="img"
    src="https://source.unsplash.com/user/willianjusten/1042x580"
  />
  <div
    class=""
  >
    <h2
      class=""
    >
      Defy death
    </h2>
    <h3
      class=""
    >
      Play the new CrashLands season
    </h3>
    <a
      class="c0"
      href="/games/defy-death"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</main>
`;
